src/osemgrep/cli_scan/Scan_subcommand.ml

```diff
@@ -91,8 +91,10 @@ let exit_code_of_errors ~strict (errors : OutJ.core_error list) : Exit_code.t =
  * to avoid having the output of multiple child processes interwinded, hence
  * the use of Unix.lockf below.
  *)
-let file_match_results_hook (conf : Scan_CLI.conf) (rules : Rule.rules)
-    (_file : Fpath.t) (match_results : Core_result.matches_single_file) : unit =
+
```

Comment:
@ajbt200128 did you see the comment above? That's a very important caveat. The hook is run in parallel in the child process created by Parmap.

Reply:
yea that's actually why were going for this approach. initially I wanted to add something like a --limit flag, but coordinating that across the parmap processes seemed pretty hacky and difficult. So this is ok to me